auth/auth-router.js

```diff
@@ -0,0 +1,57 @@
+const router = require('express').Router();
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+const secret = require('../config/secrets.js')
+const Users = require('../users/users-model.js');
+
+// for endpoints beginning with /api/auth
+router.post('/register', (req, res) => {
+  let user = req.body;
+  const hash = bcrypt.hashSync(user.password, 10); // 2 ^ n
+  user.password = hash;
+
+  Users.add(user)
+    .then(saved => {
+      res.status(201).json(saved);
+    })
+    .catch(error => {
+      res.status(500).json(error);
+    });
+});
+
+router.post('/login', (req, res) => {
+  let { username, password } = req.body;
+
+  Users.findBy({ username })
+    .first()
+    .then(user => {
+      if (user && bcrypt.compareSync(password, user.password)) {
+        const token = generateToken(user);
```

Comment:
Be careful about doing this, as this will send the hashed password back to the user in the token, which could in turn present a security risk should their local browser/cache ever be compromised. Consider tailoring `user` to not include the password hash.